libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java

```diff
@@ -1452,4 +1467,89 @@ public static String markMediaFailed(Context context, @NonNull String postConten
         }
         return postContent;
     }
+
+    public static boolean hasMediaItemsMarkedUploading(Context context, @NonNull String postContent) {
+        return hasMediaItemsMarkedWithTag(context, postContent, ATTR_STATUS_UPLOADING);
+    }
+
+    public static boolean hasMediaItemsMarkedFailed(Context context, @NonNull String postContent) {
+        return hasMediaItemsMarkedWithTag(context, postContent, ATTR_STATUS_FAILED);
+    }
+
+    private static boolean hasMediaItemsMarkedWithTag(Context context, @NonNull String postContent, String tag) {
+        // fill in Aztec with the post's content
+        AztecText content = new AztecText(context);
+        content.fromHtml(postContent);
+
+        // get all items with "uploading" class
+        AztecText.AttributePredicate uploadingPredicate = getPredicateWithClass(tag);
+
+        List<AztecAttributes> attrs = content.getAllElementAttributes(uploadingPredicate);
+
+        return (attrs != null && !attrs.isEmpty());
+    }
+
+    public static String resetUploadingMediaToFailed(Context context, @NonNull String postContent) {
+        // fill in Aztec with the post's content
+        AztecText content = new AztecText(context);
+        content.fromHtml(postContent);
+
+        // get all items with "failed" class, and make sure they are still failed
+        // i.e. if they have a local src, then they are failed.
+        resetMediaWithStatus(content, ATTR_STATUS_FAILED);
+        // get all items with "uploading" class, and make sure they are either already uploaded
+        // (that is, they have a remote src), and mark them "failed" if not.
+        resetMediaWithStatus(content, ATTR_STATUS_UPLOADING);
+
+        // re-set the post content
+        postContent = content.toHtml(false);
+        return postContent;
+    }
+
+    private static void resetMediaWithStatus(AztecText content, String status) {
+        // get all items with "uploading" class
+        AztecText.AttributePredicate uploadingPredicate = getPredicateWithClass(status);
```

Comment:
Same comment of above ^ - not sure which of these 2 implementations is better (and also faster).